### PR TITLE
[stackdriver adapter] Use 64bit trace-id based sampler

### PR DESCRIPTION
**Pull request by douglas-reid**

### CODE REVIEW

File: mixer/adapter/stackdriver/trace/trace.go [hunk 1]
```diff
@@ -42,6 +43,25 @@ type (
...
+	} else if fraction >= 1 {
+		return trace.AlwaysSample()
+	}
+
+	traceIDUpperBound := uint64(fraction * (1 << 63))
+	return trace.Sampler(func(p trace.SamplingParameters) trace.SamplingDecision {
+		if p.ParentContext.IsSampled() {
+			return trace.SamplingDecision{Sample: true}
+		}
+		x := binary.BigEndian.Uint64(p.TraceID[8:])
```
Comment (bianpengyuan) [inline mixer/adapter/stackdriver/trace/trace.go]: I think this needs to be shift 1 bit to the right for comparison: https://github.com/census-instrumentation/opencensus-go/blob/master/trace/sampling.go#L55

Comment (douglas-reid) [reply mixer/adapter/stackdriver/trace/trace.go]: I, too, was initially under that impression based on the source.  But that's why I added the `TestSampler` test.  If we shift right, we get double the expected samples for any specified rate (as we are halving the sample values).

Comment (douglas-reid) [reply mixer/adapter/stackdriver/trace/trace.go]: Whoops. My test code was bad. Fixed.

**Comment by douglas-reid** [comment]
Closing. The proper way to address this is to never use the SD sampler (set it to 1) and rely on the Pilot-configured sampling rate.

**Pull request closed without merging**